src/components/Modals/NewFileModal.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import { BsFillCheckCircleFill } from 'react-icons/bs';
+import toast from 'react-hot-toast';
+
+import { ConfirmModal } from './index';
+import examples from '../../examples';
+import { EditorService } from '../../services';
+import state from '../../state';
+
+export const NewFileModal: React.FunctionComponent = () => {
+  const sidebarState = state.useSidebarState();
+  const [selectedTemplate, setSelectedTemplate] = useState({ title: '', template: '' });
+
+  const handleSubmit = () => {
+    EditorService.updateState({ content: selectedTemplate.template, updateModel: true });
+    const panels = state.sidebar.panels;
+    panels.merge({
+      newFile: false,
+    });
+
+    toast.success(
+      <div>
+        <span className="block text-bold">
+          Succesfully reused the {`"${selectedTemplate.title}"`} template.
+        </span>
+      </div>
+    );
+  };
+
+  const realLifeExamples = examples.filter((template) => template.type === 'real-example');
+  const templates = examples.filter((template) => template.type === 'protocol-example');
+
+  return (
+    <ConfirmModal containerClassName="sm:max-w-6xl" onCancel={() => sidebarState.panels.newFile.set(false)} title="AsyncAPI Templates - Start with our template examples" confirmText="Use Template" confirmDisabled={false} show={true} onSubmit={handleSubmit}>
+      <div className="flex content-center justify-center">
+        <div className="w-full  overflow-auto space-y-8 ">
+          <div>
+            <span className="uppercase text-gray-800 text-sm underline font-bold">Templates</span>
+            <div className="grid grid-cols-3 gap-4 py-4">
+              {templates.map(({ title, description: Description, template }) => {
+                const isSelected = selectedTemplate.title === title;
+                const containerStyles = isSelected ? 'border-pink-500' : 'border-gray-200';
+                const textStyles = isSelected ? 'text-pink-600' : 'text-gray-600';
+
+                return (
+                  <button
+                    onClick={() => setSelectedTemplate({ title, template })}
+                    key={title}
+                    className={`group text-left flex flex-col cursor-pointer rounded-lg p-4 pb-6 border-2 hover:border-pink-500 ${containerStyles}`}
+                  >
+                    <div className="flex justify-between w-full">
+                      <span className={`block text-md  font-bold leading-0 group-hover:text-pink-600 ${textStyles} `}>{title}</span>
+                      {isSelected && <BsFillCheckCircleFill className="w-5 h-5 text-pink-600" />}
+                    </div>
+                    <span className="block text-sm text-gray-500 font-light mt-1 group-hover:text-gray-900">
+                      <Description />
+                    </span>
+                  </button>
+                );
+              })}
```

Comment:
Hi @boyney123 🙂, I took a look at the Netlify preview, and this new modal looks amazing! I noticed SonarCloud found duplicate lines so maybe we could separate this logic into its own component, what do you think?

Something like this:
```ts
interface TemplateListItemProps {
  title: string;
  description: React.FunctionComponent;
  template: string;
}

const TemplateListItem: React.FunctionComponent<TemplateListItemProps> = ({
  title,
  description: Description,
  template,
  isSelected
}) => {
  const containerStyles = isSelected ? 'border-pink-500' : 'border-gray-200';
  const textStyles = isSelected ? 'text-pink-600' : 'text-gray-600';

  return (
    <button
      onClick={() => setSelectedTemplate({ title, template })}
      key={title}
      className={`group text-left flex flex-col cursor-pointer rounded-lg p-4 pb-6 border-2 hover:border-pink-500 ${containerStyles}`}
    >
      <div className="flex justify-between w-full">
        <span className={`block text-md font-bold leading-0 group-hover:text-pink-600 ${textStyles}`}>{title}</span>
        {isSelected && <BsFillCheckCircleFill className="w-5 h-5 text-pink-600" />}
      </div>
      <span className="block text-sm text-gray-500 font-light mt-1 group-hover:text-gray-900">
        <Description />
      </span>
    </button>
  );
};
```

Also, it seemed like the HTML used in the `realLifeExamples` case had an extra `border-gray-200`, and if it was selected it had these two classes:
![imagem](https://user-images.githubusercontent.com/18630253/146674104-298db067-4c24-4ae3-9231-a3e14016519f.png)

Reply:
Just looking at this now, thank you @BOLT04 , I used your example and it worked fine (with a few tweaks!). Thanks for that.

Also the border duplication is removed, nice catch!